### DISCUSSION TITLE
Plot pyvista in spyder

### DISCUSSION THREAD

**Discussion by byba1990** [pyvista/pyvista#3220] [Q&A]
Hi all,

I am trying to use pyvista with spyder. There is a possibility to show the plots inline? I tried to plot a mesh with these options:

```
mesh.plot(color='blue', show_edges=True, show_bounds='True', interactive=False, 
            jupyter_backend=False, notebook=False)
```

but I obtained a plot shown in an OpenGL window. I cannot also close this window (maybe a bug?).
I would like to have the possibility to have plots without this problem displayed also in this way (with the OpenGL window), in addition to have the possibility to get static plots inline, on the spyder console.
Any suggestion, please?
Thanks in advance for your time.
Regards,

Christian

**Comment by akaszynski** [comment]
Hello,

I'm guessing that you're seeing this issue on MacOS or with VTK <9.1.0. Could you please paste a `pyvista.Report()` here?

**Reply by byba1990** [reply]
Dear @akaszynski ,

Thanks for your answer.
I'm working on a GNU/Linux system.
Here below, what pyvista.Report() shows:


--------------------------------------------------------------------------------
  Date: Sun Aug 28 17:08:24 2022 IDT

                OS : Linux
            CPU(s) : 56
           Machine : x86_64
      Architecture : 64bit
               RAM : 125.0 GiB
       Environment : IPython
       File system : xfs
        GPU Vendor : Mesa/X.org
      GPU Renderer : llvmpipe (LLVM 13.0.1, 256 bits)
       GPU Version : 4.5 (Core Profile) Mesa 21.3.4

  Python 3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:35:26) [GCC
  10.4.0]

           pyvista : 0.36.1
               vtk : 9.1.0
             numpy : 1.23.2
           imageio : 2.21.1
           appdirs : 1.4.4
            scooby : 0.5.12
        matplotlib : 3.5.3
             PyQt5 : 5.12.3
           IPython : 7.33.0
             scipy : 1.9.0
              tqdm : 4.64.0
        jupyterlab : 3.4.5
--------------------------------------------------------------------------------`

Thanks for your time.
Regards,

Christian

**Comment by byba1990** [comment]
Hi, 

I'm still having this problem. I also would like to know if it is possible to avoid plotting the image on the screen, with the intention to only save the file in png format. If I use this setting:

`mesh.plot( cpos='xy', show_edges=True, interactive=False, jupyter_backend=False, screenshot="mesh.png")
` 
I'm able to save the image, with still the plot shown on the screen.

Thanks in advance for your time.
Best regards,

Christian

**Reply by banesullivan** [reply]
Sounds like you want `off_screen=True` which will plot and save the screenshot without any interaction.
